### ISSUE TITLE
ZipInputStream did't get all entries for this file

### ISSUE THREAD

**Issue by foobnix** [srikanth-lingala/zip4j#143]

**Comment by foobnix** [comment]
[Fairfax, John - Summary Justice.epub.zip](https://github.com/srikanth-lingala/zip4j/files/4191150/Fairfax.John.-.Summary.Justice.epub.zip)

**Comment by srikanth-lingala** [comment]
Which files did zip4j miss to unzip? I used The Unarchiver on mac to unzip your zip file, and the content which was extracted by this tool was the same as the content extracted by zip4j. I also checked with java's inbuilt tool and it was also the same here as well. Which tool did you use to unzip the zip file?

**Comment by foobnix** [comment]
Please test this file with ZipInputStream and ZipFile

```
public static void test(File file) throws IOException {
        //case 1
        ZipInputStream zis = new ZipInputStream(new FileInputStream(file));
        LocalFileHeader nextEntry = zis.getNextEntry();
        while (nextEntry != null) {
            LOG.d("Zip-1", nextEntry.getFileName());
            nextEntry = zis.getNextEntry();
        }
        zis.close();
        //case 2
        ZipFile zipFile = new ZipFile(file);
        List<FileHeader> fileHeaders = zipFile.getFileHeaders();
        for (FileHeader h : fileHeaders) {
            LOG.d("Zip-2", nextEntry.getFileName());
        }
    }
```
result is different:

D/DEBUG   ( 1935): Zip-1|mimetype|
D/DEBUG   ( 1935): Zip-1|META-INF/|
D/DEBUG   ( 1935): Zip-2|mimetype|
D/DEBUG   ( 1935): Zip-2|META-INF/|
D/DEBUG   ( 1935): Zip-2|META-INF/container.xml|
D/DEBUG   ( 1935): Zip-2|content.opf|
D/DEBUG   ( 1935): Zip-2|cover.jpeg|
D/DEBUG   ( 1935): Zip-2|images/|
D/DEBUG   ( 1935): Zip-2|images/00001.gif|
D/DEBUG   ( 1935): Zip-2|page_styles.css|
D/DEBUG   ( 1935): Zip-2|stylesheet.css|
D/DEBUG   ( 1935): Zip-2|text/|
D/DEBUG   ( 1935): Zip-2|text/part0000.html|
D/DEBUG   ( 1935): Zip-2|text/part0001.html|
D/DEBUG   ( 1935): Zip-2|text/part0002.html|
D/DEBUG   ( 1935): Zip-2|text/part0003.html|
D/DEBUG   ( 1935): Zip-2|text/part0004.html|
D/DEBUG   ( 1935): Zip-2|text/part0005.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_000.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_001.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_002.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_003.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_004.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_005.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_006.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_007.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_008.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_009.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_010.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_011.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_012.html|
D/DEBUG   ( 1935): Zip-2|text/part0006_split_013.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_000.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_001.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_002.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_003.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_004.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_005.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_006.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_007.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_008.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_009.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_010.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_011.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_012.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_013.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_014.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_015.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_016.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_017.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_018.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_019.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_020.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_021.html|
D/DEBUG   ( 1935): Zip-2|text/part0007_split_022.html|
D/DEBUG   ( 1935): Zip-2|text/part0008_split_000.html|
D/DEBUG   ( 1935): Zip-2|text/part0008_split_001.html|
D/DEBUG   ( 1935): Zip-2|text/part0008_split_002.html|
D/DEBUG   ( 1935): Zip-2|text/part0008_split_003.html|
D/DEBUG   ( 1935): Zip-2|text/part0008_split_004.html|
D/DEBUG   ( 1935): Zip-2|text/part0008_split_005.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_000.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_001.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_002.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_003.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_004.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_005.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_006.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_007.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_008.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_009.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_010.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_011.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_012.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_013.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_014.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_015.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_016.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_017.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_018.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_019.html|
D/DEBUG   ( 1935): Zip-2|text/part0009_split_020.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_000.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_001.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_002.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_003.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_004.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_005.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_006.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_007.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_008.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_009.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_010.html|
D/DEBUG   ( 1935): Zip-2|text/part0010_split_011.html|
D/DEBUG   ( 1935): Zip-2|text/part0011.html|
D/DEBUG   ( 1935): Zip-2|text/part0012.html|
D/DEBUG   ( 1935): Zip-2|titlepage.xhtml|
D/DEBUG   ( 1935): Zip-2|toc.ncx|

**Comment by srikanth-lingala** [comment]
Thanks for the sample code. I was able to reproduce it. Working on a fix.

**Comment by srikanth-lingala** [comment]
Fixed in v2.3.2 released today
